tests/Nelmio/Alice/Fixtures/Builder/Reference.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/*
+ * This file is part of the Alice package.
+ *  
+ * (c) Nelmio <hello@nelm.io>
+ *  
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Nelmio\Alice\Fixtures\Builder\Methods;
+
+use Nelmio\Alice\Fixtures\Fixture;
+
+/**
+ * List a list of fixture reference, with the expected result. The list is flagged to be able to easily extract some
+ * parts of it. This class aims at centralising all the possible fixture tested to avoid too much duplication.
+ */
+final class Reference
```

Comment:
Notable changes with the system prior to that PR:

- `user_{alice, bob,}` previously was building a reference named `user_{alice, bob,}`. Now builds that as a list, i.e. result in `user_alice` and `user_bob`. A deprecation warning is also thrown to warn the user that the list is poorly formatted and an exception will be thrown in v3.
- `user_{, alice, bob}`: same as previous case.
- `user_{0..2}`: value for `<current()>` were respectively `'0'`, `1`, `2`. Now are all strings as states the phpdoc. Changed in #339.
- `user_{0....2}`: was generating only one fixture named `user_{0....2}`; now is equivalent to `user_{0...2}`
- `user_{2...0}`: as reported in #358 was generating 4 fixtures... now is equivalent to `0...2` which result in `user_0`, `user_1` and `user_2`
- `user_{2...2}`: was generating two fixtures `user_1` and `user_3`, now doesn't build any (the segment is `[2;2[` so contains no element)
- `user_{0.2}`: was generating a fixture named `user_{0.2}`. No doesn't generate any.
- `user_{2..}`: was generating a fixture named `user_{2..}`. No doesn't generate any.
- `user_{-1..2}`: was generating a fixture named `user_-1..2`. No doesn't generate any. Same goes for all ranges containing a negative number

You can guess that they required to drastically change the behaviour of the Builder methods, which *is* a BC break. However as it's a fairly low level thing and this is necessary to fix those bugs, I reckon it can be changed?

Makes me thing that those Methods should be annotated as internal.